### DISCUSSION TITLE
how to reset all values in an array?

### DISCUSSION THREAD

**Discussion by TheColorRed** [slint-ui/slint#6659] [Q&A] [answered]
I am looking to loop over any array and reset a setting back to `false`, is this sported, or is there some kind of trick that I am not thinking of?

here is some pseudo code (untested):

```slint
export struct Example {
  selected: bool
}

export component AppWindow inherits Window {
  property <int> current: 1;
  property <[Example]> items: [
    { selected: false; }
    { selected: false; }
    { selected: false; }
  ]

  VerticalLayout {
    for x in items: Rectangle {
      background: items[current].selected ? #ffffff58 : #ffffff2e;
      height: 100px;
      ta := TouchArea {
        clicked => {
            for i in items { i.selected = false; } // Reset the values to false
            items[current].selected = true;
        }
      }
    }
  }
}
```

**Comment by ogoffart** [comment, marked as answer]
This is the kind of things that are part of the logic and which needs to be done in the programming language at the moment.
Add a callback to your AppWindow, and this callback can be set to iterate over the model and unselect everything.

**Reply by TheColorRed** [reply]
Dang. I was hoping something was built in. This will work though.
